extensions/data-plane/data-plane-google-storage/src/main/java/org/eclipse/edc/connector/dataplane/gcp/storage/GcsDataSourceFactory.java

```diff
@@ -44,13 +44,18 @@ public boolean canHandle(DataFlowRequest request) {
 
     @Override
     public @NotNull Result<Boolean> validate(DataFlowRequest request) {
+        return validateRequest(request).map(it -> true);
+    }
+
+    @Override
+    public @NotNull Result<Void> validateRequest(DataFlowRequest request) {
         var source = request.getSourceDataAddress();
-        return validation.apply(source).map(it -> true);
+        return validation.apply(source).map(it -> null);
```

Comment:
the `map(it -> null)` is not needed as validation function already returns a `Result<Void>`